src/type_checker/checker.rs

```diff
@@ -311,7 +311,8 @@ impl<B: Backend> TypeChecker<B> {
             ExprKind::Variable { module, name } => {
                 let qualified = FullyQualified::new(module, &name.value);
 
-                if is_type(&name.value) {
+                // generic parameter should be checked as a local variable
+                if is_type(&name.value) && !is_generic_parameter(&name.value) {
```

Comment:
can you expand here? I'm not sure I understand what is happening here, you can't declare a generic variable at the roof of a module so I'm not sure how you could import it using `mod::Thing`

Reply:
The current string patterns for the `is_type` and `is_generic_parameter` is overlapped.

Because `NN` can be also seen as a type string. The use `!is_generic_parameter` is to skip the branch for handling the type variable.

So I think we will have to enforce the type pattern to the camel case. Then this addition of `!is_generic_parameter` can be reverted.